plugin/restore/mysql/mysql.go

```diff
@@ -144,6 +152,156 @@ func getPITROldDatabaseName(database string, suffixTs int64) string {
 	return getSafeName(database, suffix)
 }
 
+// SyncArchivedBinlogFiles syncs the binlogs between the instance and `saveDir`,
+// but exclude latest binlog. We will download the latest binlog only when doing PITR.
+func (r *Restore) SyncArchivedBinlogFiles(ctx context.Context, instance *api.Instance, saveDir string) error {
+	binlogFilesLocal, err := ioutil.ReadDir(saveDir)
+	if err != nil {
+		return err
+	}
+
+	binlogFilesOnServer, err := r.getBinlogFilesMetaOnServer(ctx)
+	if err != nil {
+		return err
+	}
+
+	latestBinlogFileOnServer, err := r.getLatestBinlogFileMeta(ctx)
+	if err != nil {
+		return err
+	}
+
+	// compare file sizes and names to decide which files to download
+	// downloadedName maintains the name of BinlogFile slice, for example:
+	// `SHOW MASTER LOGS` return ["binlog.000001", "binlog.000002"],
+	// if we had downloaded binlog.000001 and file size match, {"binlog.000001":struct{}{}} will be involved in downloadedName.
+	downloadedName := make(map[string]struct{})
```

Comment:
Can we use map[string]bool instead? struct{} isn't quite intuitive.

Reply:
SG

Reply:
But `resultFileDir := strings.TrimRight(saveDir, "/") + "/"` is to ensure that the path must end with '/'. For example:
```bash
> mysqlbinlog binlog.000001 --result-file=/tmp
```
On my machine, the mysqlbinlog would not save binlog.000001 in /tmp. Instead, save binlog.000001 as /tmpbinlog.000001. @d-bytebase